### DISCUSSION TITLE
Is there any way to perform parallel tools calling?

### DISCUSSION THREAD

**Discussion by jakexia72** [instructor-ai/instructor-js#122] [Q&A] [answered]
I see in the docs that we're limited to one response model right now for simplicity sake but perhaps it's possible to allow response-model be of type zod.array or zod.union? It's my first time using zod; please let me know if I'm understanding it wrong! 

It appears to be possible with the python/pydantic version as seen in this [example](https://github.com/jxnl/instructor/blob/main/examples/parallel/run.py).

**Comment by jakexia72** [comment, marked as answer]
actually I just found [this](https://gist.github.com/jxnl/69d2e8a594ef6de8d56867f3e7ef2ea6)... I will just create a parent object instead.

**Reply by oleksii-lukin** [reply]
is this the right link? It's some python script not related to zod? I also interested in same question of returning multiple objects and looking at debug data model returns me an array of objects, but instructor.chat.completions.create only returns single object and don't accept zod array as schema
